internal/pkg/gateway/commit/notifier_test.go

```diff
@@ -0,0 +1,285 @@
+/*
+Copyright 2021 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/core/ledger"
+	"github.com/hyperledger/fabric/internal/pkg/gateway/commit"
+	"github.com/hyperledger/fabric/internal/pkg/gateway/commit/mock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+//go:generate counterfeiter -o mock/notificationsupplier.go --fake-name NotificationSupplier . NotificationSupplier
+
+func TestNotifier(t *testing.T) {
+	newTestNotifier := func(commitSend <-chan *ledger.CommitNotification) *commit.Notifier {
+		notificationSupplier := &mock.NotificationSupplier{}
+		notificationSupplier.CommitNotificationsReturnsOnCall(0, commitSend, nil)
+		notificationSupplier.CommitNotificationsReturns(nil, errors.New("unexpected call of CommitNotificationChannel"))
+
+		return commit.NewNotifier(notificationSupplier)
+	}
+
+	t.Run("NewNotifier with nil ledger returns nil", func(t *testing.T) {
+		result := commit.NewNotifier(nil)
+		require.Nil(t, result)
+	})
+
+	t.Run("Notify", func(t *testing.T) {
+		t.Run("returns error from notification supplier", func(t *testing.T) {
+			notificationSupplier := &mock.NotificationSupplier{}
+			notificationSupplier.CommitNotificationsReturns(nil, errors.New("MY_ERROR"))
+			notifier := commit.NewNotifier(notificationSupplier)
+			defer notifier.Close()
+
+			_, err := notifier.Notify(context.Background().Done(), "CHANNEL_NAME", "TX_ID")
+
+			require.ErrorContains(t, err, "MY_ERROR")
+		})
+
+		t.Run("delivers notification for matching transaction", func(t *testing.T) {
+			commitSend := make(chan *ledger.CommitNotification, 1)
+			notifier := newTestNotifier(commitSend)
+			defer notifier.Close()
+
+			commitReceive, err := notifier.Notify(context.Background().Done(), "CHANNEL_NAME", "TX_ID")
+			require.NoError(t, err)
+
+			commitSend <- &ledger.CommitNotification{
+				BlockNumber: 1,
+				TxIDValidationCodes: map[string]peer.TxValidationCode{
+					"TX_ID": peer.TxValidationCode_MVCC_READ_CONFLICT,
+				},
+			}
+			actual := <-commitReceive
```

Comment:
Be aware that we have historically had issues with are channel receives in tests. They have no timeout so when something goes awry, the tests hang for 20 minutes. You don't need to do anything right now but these patterns require consideration as you build out the code.

Reply:
I like to see my tests fail before I make them pass so I've seen this a lot. I always run my unit tests with `go test`'s built-in `-timeout` parameter so they don't block for more than a couple of seconds before failing. Is it worth including the `-timeout` option in the main test run or is it necessary to write explicit timeout handling code in all the unit tests?